dashboard/app/models/plc/script_completion_task.rb

```diff
@@ -21,4 +21,17 @@ class Plc::ScriptCompletionTask < Plc::Task
   def script
     script_id ? Script.find(script_id) : nil
   end
+
+  def self.check_for_script_completion user_script
+    #Get all the task assignments for the user for script level tasks
+
+    #I'm not wild about the joinery here - we can definitely save time by adding denormalizing user_id on the
+    #enrollment_task_assignment table. The number of script completions tasks associated with a user won't be that
+    #high, so it's okay to select all of them and pick the ones we want
+    user_script.user.plc_task_assignments.joins(:plc_task).
+                          where('plc_tasks.type': 'Plc::ScriptCompletionTask').
+                          includes(:plc_task).
+                          select {|x| x.plc_task.script_id == user_script.script_id}.
+                          map(&:complete_assignment)
```

Comment:
`map` could be changed to `each` to be more clear that the return value of this method chain is not meaningful and/or used downstream. (And `complete_assignment` could be renamed to `complete_assignment!` to make it more clear that calling the method produces a side-effect rather than returning a value)